sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -831,7 +858,9 @@ public CosmosClientBuilder clientTelemetryConfig(CosmosClientTelemetryConfig tel
     public CosmosAsyncClient buildAsyncClient() {
         validateConfig();
         buildConnectionPolicy();
-        return new CosmosAsyncClient(this);
+        CosmosAsyncClient cosmosAsyncClient = new CosmosAsyncClient(this);
+        cosmosAsyncClient.openConnectionsAndInitCaches();
```

Comment:
where the `proactiveContainerInitConfig` is used?

Reply:
The `CosmosClient` and `CosmosAsyncClient` constructor takes the `CosmosClientBuilder` instance as an argument which has `ProactiveContainerInitConfig` encapsulated. It is internally set from `CosmosClientBuilder` to CosmosClient and `CosmosAsyncClient`.

Reply:
ah got it - sorry missed the setting part